### PR TITLE
feat(gitlab)!: use preprod-${CI_PROJECT_NAME} ingress for preprod-dev2

### LINKED ISSUE #142: preprod envs urls
Probleme avec les urls gitlab en preprod et composant app

exemple sur https://github.com/SocialGouv/cdtn-veille-git qui ne précise pas de hostname spécifique

Au merge sur master :
 - OK : un env est crée pour master sur https://master-dev2-cdtn-veille-git.dev2.fabrique.social.gouv.fr
 - **NOK** : une release est délenchée, laquelle crée un env : 
     - qui apparait dans l'env gitlab "preprod-dev2" en tant que https://preprod-cdtn-veille-git.dev2.fabrique.social.gouv.fr/ mais l'ingress réel est sur v1-0-0-cdtn-veille-git.dev2.fabrique.social.gouv.fr
 - OK : prod créée sur https://cdtn-veille-git.fabrique.social.gouv.fr/

### PR DESCRIPTION
**Pull request by revolunet**
use a fixed url for preprod ingress, when `CI_ENVIRONMENT_NAME=preprod-dev2`

also add more contextual e2e tests variables

also remove nginx from simple e2e template to not conflict with "app"

fix #142

### CODE REVIEW

File: e2e/templates/simple/kosko generate/__tests__/--env preprod.ts [hunk 1]
```diff
@@ -22,6 +22,8 @@ test(
           ...gitlabEnv,
           CI_COMMIT_TAG: "v1.2.3",
           KUBE_NAMESPACE: "sample-next-app-24-preprod-dev2",
+          CI_ENVIRONMENT_NAME: "preprod-dev2",
+          CI_ENVIRONMENT_SLUG: "preprod-dev2",
```
Comment (douglasduteil) [inline e2e/templates/simple/kosko generate/__tests__/--env preprod.ts:26]: @revolunet can you sort the object keys ?

File: e2e/templates/simple/kosko generate/__tests__/--env prod.ts [hunk 2]
```diff
@@ -18,7 +18,14 @@ test(
...
       {
         cwd,
-        env: { ...gitlabEnv, CI_COMMIT_TAG: "v1.2.3", PRODUCTION: "true" },
+        env: {
+          ...gitlabEnv,
+          CI_COMMIT_TAG: "v1.2.3",
+          PRODUCTION: "true",
+          CI_ENVIRONMENT_NAME: "prod2",
+          CI_ENVIRONMENT_SLUG: "prod2",
+          KUBE_INGRESS_BASE_DOMAIN: "fabrique.social.gouv.fr",
```
Comment (douglasduteil) [inline e2e/templates/simple/kosko generate/__tests__/--env prod.ts:27]: @revolunet can you sort the object keys ?

File: e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env dev.ts.snap [hunk 3]
```diff
@@ -153,130 +153,6 @@ spec:
...
-              serviceName: app
-              servicePort: 80
-            path: /
-  tls:
-    - hosts:
-        - e2e-branch-42-sample-next-app.dev.fabrique.social.gouv.fr
-      secretName: wildcard-crt
-apiVersion: extensions/v1beta1
-kind: Ingress
----
```
Comment (douglasduteil) [inline e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env dev.ts.snap:279]: No more deployment, service and ingress ?

File: e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env preprod.ts.snap [hunk 4]
```diff
@@ -156,132 +156,8 @@ kind: Service
...
-        - v1-2-3-sample-next-app.dev.fabrique.social.gouv.fr
-      secretName: wildcard-crt
-apiVersion: extensions/v1beta1
-kind: Ingress
----
-metadata:
-  annotations:
-    app.gitlab.com/app: socialgouv-sample-next-app
-    app.gitlab.com/env: e2e-branch-42
-    app.gitlab.com/env.name: e2e-branch-dev2
```
Comment (douglasduteil) [inline e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env preprod.ts.snap:284]: No more deployment, service and ingress ?

File: e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env prod.ts.snap [hunk 5]
```diff
@@ -138,140 +135,13 @@ kind: Service
...
-        - sample-next-app.dev.fabrique.social.gouv.fr
-      secretName: app-crt
-apiVersion: extensions/v1beta1
-kind: Ingress
----
-metadata:
-  annotations:
-    app.gitlab.com/app: socialgouv-sample-next-app
-    app.gitlab.com/env: e2e-branch-42
-    app.gitlab.com/env.name: e2e-branch-dev2
```
Comment (douglasduteil) [inline e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env prod.ts.snap:268]: No more deployment, service and ingress ?

**Review by douglasduteil — CHANGES_REQUESTED** [review]

File: e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env dev.ts.snap (hunk 3 above)
Comment (revolunet) [reply e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env dev.ts.snap:279]: this is the nginx container removed from the e2e template

File: e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env prod.ts.snap (hunk 5 above)
Comment (revolunet) [reply e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env prod.ts.snap:268]: this is the nginx container removed from the e2e template

File: e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env preprod.ts.snap (hunk 4 above)
Comment (revolunet) [reply e2e/templates/simple/kosko generate/__tests__/__snapshots__/--env preprod.ts.snap:284]: this is the nginx container removed from the e2e template

**Pull request merged**